lib/src/parse/parser.dart

```diff
@@ -74,16 +74,21 @@ class Parser {
   // ## Tokens
 
   /// Consumes whitespace, including any comments.
+  ///
+  /// Set `consumeNewlines` to true when a statement could not end, so newlines
+  /// can be consumed. Only used in the indented syntax.
+  ///
+  /// TODO: Bikeshed the variable name
   @protected
-  void whitespace() {
+  void whitespace({bool consumeNewlines = false}) {
```

Comment:
@nex3 I'd appreciate a quick check that this approach is in the right direction. Essentially, I'm looking at every invocation of `whitespace()`, and determining whether a statement could end, in which case `consumeNewlines` should be false.

(For the WIP, I'm explicitly setting `consumeNewlines: false` in places I've covered, just to track progress, but those would be cleaned up.)

So far, this has covered most of my needs, with a few edge cases inside `_expression`.

I also want to check my assumption that the AST parsing classes should mirror these changes- for instance, should `/lib/src/parse/media_query.dart` mirror the changes in `/lib/src/parse/stylesheet.dart:_mediaQuery`?

Reply:
It's a little unfortunate that we'll have to remember to consider each call to `whitespace()` in the future to determine whether or not it needs this flag, but I don't see a clearly better way to handle this. If you want to require every call site to specify this explicitly, you can write:

```suggestion
  void whitespace({required bool consumeNewlines}) {
```

> I also want to check my assumption that the AST parsing classes should mirror these changes- for instance, should `/lib/src/parse/media_query.dart` mirror the changes in `/lib/src/parse/stylesheet.dart:_mediaQuery`?
>
> There are also some `parse*()` functions in `stylesheet.dart` that seem to only be used by the AST, and mirror other functions- for instance `parseArgumentDeclaration` and `_argumentDeclaration`, or `parseUseRule` and `_useRule`. What are these used for, and (if the changes need to be applied there as well) how would I test them?

The other parser classes are usually used to do secondary parses of interpolated constructs. Since they're not being parsed at the stylesheet level, whitespace should never be meaningful for them, so they should always allow newlines.

Reply:
Thanks!

I was initially thinking of using `whitespace(consumeNewlines: false)` temporarily to track where I've checked, and then going back and removing the default value. But you're right that we want future changes to know they have to be intentional about this as well, and I'll make it required once I've made the initial pass.